torch/_dynamo/guards.py

```diff
@@ -1776,6 +1776,9 @@ def CLOSURE_MATCH(self, guard: Guard):
             self.FUNCTION_MATCH(guard)
 
     def BUILTIN_MATCH(self, guard: Guard):
+        # don't support this in serialization because it uses unsupported FUNCTION_MATCH
+        if self.serialization_mode == "save":
+            raise RuntimeError("BUILTIN_MATCH guard cannot be serialized.")
```

Comment:
Any reason the check can't be in FUNCTION_MATCH.

Builtins should be easy to support since there is a finite list of them.

Reply:
Sounds good, I removed this check and let it pass through to the FUNCTION_MATCH error instead.

Does it makes sense in this PR to update the BUILTIN_MATCH guard logic to support the list of builtins instead of using a FUNCTION_MATCH guard or should that be done in a follow-up?